### ISSUE TITLE
Some content becomes unreadable with text enlarge

### ISSUE THREAD

**Issue by karlgroves** [WordPress/gutenberg#15351]
Some content becomes unreadable with text enlarge

-   **Severity:** Medium
-   **Affected Populations:**
    -   Low-Vision
-   **Platform(s):**
    -   All / Universal
-   **Components affected:**
    -   Block Editing

#### Issue description

Users who need to increase the text size will find that some block
content overlaps its containers, and other content.

Specifically, the "Change Block Type" dropdown does not grow to
encompass the Type blocks, causing them to overflow the dropdown
container. Without the container background, it is difficult to read the
block type name against the editable page below, when the item is not
being hovered over (since hovering adds a background). The words of the
type also spill over the text of adjacent types, making the text
unreadable.

Some of the actionable buttons, such as setting a heading level within
the toolbar, also become difficult to read due to text overlapping.

The ability to resize text is essential for users with low-vision, and
may be helpful for users who have a cognitive disability. Catering to
zoom alone is not sufficient because the browser's font-size may be
increased independently of zoom level.

##### Issue Code

```
    .editor-block-types-list__list-item {


        ...


        width: 33.33%;


        ...


    }


    ...


    /* hx buttons */


    <button type="button" aria-label="Heading 2" data-subscript="2" aria-pressed="true" class="...">


        <svg ..." width="20" height="20" viewBox="0 0 20 20">...</svg>


    </button>


    ...


    .components-toolbar__control.components-button[data-subscript]::after {


        content: attr(data-subscript);


        ...


        font-size: 13px;


        font-weight: 600;


        line-height: 12px;


        position: absolute;


        right: 8px;


        bottom: 10px;


    }
```

#### Remediation Guidance

Keep width and height minimums to 44px or greater. Set minimum sizes
using `min-width` and `min-height`, rather than fixed sizes using
`width` and `height`, to allow containers to expand with increases
in their content size.

Set SVG sizes in units that will scale with text, such as `em` or
`rem`.

Avoid using absolute positioning for text that is close enough to other
text to overlap at larger font sizes. Absolutely-positioned elements are
outside the layout flow, and cannot be adjusted for by the layout of
other elements.

##### Recommended Code

```
    .editor-block-types-list__list-item {


        ...


        min-width: 33.33%;


        ...


    }


    ...


    /* hx buttons */


    <button type="button" aria-label="Heading 2" data-subscript="2" aria-pressed="true" class="...">


        <svg ..." viewBox="0 0 20 20">...</svg>


    </button>


    .components-toolbar__control.components-button:not(:disabled).is-active>svg {


        ...


        height: 1em;


        width: 1em;


    }


    ...


    .components-toolbar__control.components-button[data-subscript]::after {


        content: attr(data-subscript);


        ...


    }

```

#### Relevant standards

-   [1.4.4 Resize text (Level
    AA)](https://www.w3.org/TR/WCAG20/#visual-audio-contrast-scale)

**Note:** This issue may be a duplicate with other existing accessibility-related bugs in this project. This issue comes from the Gutenberg accessibility audit, performed by [Tenon](https://www.tenon.io) and funded by [WP Campus](https://wpcampus.org/). This issue is GUT-96 in Tenon's report

**Comment by melchoyce** [comment]
Screenshots from full report:

![image](https://user-images.githubusercontent.com/2846578/57185801-36c48580-6e87-11e9-9b0f-8f3030c40134.png)
![image](https://user-images.githubusercontent.com/2846578/57185803-3cba6680-6e87-11e9-9bf5-cd876d4527ca.png)

**Comment by afercia** [comment]
Discussed during today's accessibility bug scrub and agreed all the points here make perfectly sense and are actionable.

Also, some of the recommendations here should really be considered for inclusion in the design guides (e.g. avoid absolute positioning).

Note: the 44x44 pixels requirement is actually for WCAG level AAA and WordPress aims to level AA. This doesn't mean buttons would not benefit from a bigger size 🙂

**Comment by afercia** [comment]
Feedback from @bemdesign
> the 44x44 pixel requirement I'd consider more optional than a requirement - I would just recommend that when designing or developing, consider touch/pointer hit areas for folks who may have difficulty hitting smaller hit targets.

**Comment by tellthemachines** [comment]
Closing as this has been fixed already. Screenshot of view on Chrome at 400% zoom:

<img width="1468" alt="Screen Shot 2020-03-30 at 3 42 26 pm" src="https://user-images.githubusercontent.com/8096000/77875864-1f5f6980-729d-11ea-86ad-0951cf4ee1fa.png">
